### PR TITLE
COD-8 Sign Up Test

### PR DESCRIPTION
**Pull request by syedishmam**
Sign Up Form
Screen Shot 2021-11-02 at 3 47 33 PM

Sign Up Test Script
Screen Shot 2021-11-02 at 3 44 57 PM

Uses Data Provider for Login Credentials
Screen Shot 2021-11-02 at 3 45 30 PM

Utilized new BasePage method for scrolling
Screen Shot 2021-11-02 at 3 46 12 PM

### CODE REVIEW

File: src/test/java/mobiletestrunner/SignUpTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+				String email,
+				String password,
+				String mobile,
+				String zipCode
+			) throws InterruptedException {
+		Thread.sleep(3000);
+		LandingPage landing = new LandingPage(driver);
+		SignUpFormPage signUp = new SignUpFormPage(driver);
+		
+		String tosAcceptSelector = "new UiSelector().className(\"android.widget.Button\").text(\"ACCEPT\")";
```
Comment (srahman09) [inline src/test/java/mobiletestrunner/SignUpTest.java:31]: There is already a method in the LandingPage class called "acceptButton" to accept the prompt when you first open the app.

File: src/test/java/mobiletestrunner/SignUpTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		driver.findElement(MobileBy.AndroidUIAutomator(signUp.emailAddressFieldSelector)).sendKeys(email);
+		driver.findElement(MobileBy.AndroidUIAutomator(signUp.passwordFieldSelector)).sendKeys(password);
+		driver.findElement(MobileBy.AndroidUIAutomator(signUp.mobileNumberFieldSelector)).sendKeys(mobile);
+		driver.findElement(MobileBy.AndroidUIAutomator(signUp.zipCodeFieldSelector)).sendKeys(zipCode);
+		
+		signUp.scroll(500, 500, 500, 100);
+		
+		Thread.sleep(3000);
+		driver.findElement(MobileBy.AndroidUIAutomator(signUp.checkBoxButtonSelector)).click();
+
```
Comment (srahman09) [inline src/test/java/mobiletestrunner/SignUpTest.java:49]: Is it not better to put all these selectors in the signup form page. There are two methods in the base page "uiSelector" and "sendQuery" which makes it easier to put all this info in page objects and avoid being repetitive. Similar to how I did MenuPage and MenuTest

File: src/test/java/mobiletestrunner/SignUpTest.java (hunk 1 above)
Comment (syedishmam) [reply src/test/java/mobiletestrunner/SignUpTest.java:31]: This is prior to you pushing your PR to master so it's not there in this version

File: src/test/java/mobiletestrunner/SignUpTest.java (hunk 2 above)
Comment (syedishmam) [reply src/test/java/mobiletestrunner/SignUpTest.java]: It's changed

File: src/test/java/mobiletestrunner/SignUpTest.java [hunk 3]
```diff
@@ -35,19 +35,17 @@ public void signUpTest(
...
+		signUp.inputToTargetUiSelector(signUp.mobileNumberFieldSelector, mobile);
+		signUp.inputToTargetUiSelector(signUp.zipCodeFieldSelector, zipCode);
 		
 		signUp.scroll(500, 500, 500, 100);
 		
 		Thread.sleep(3000);
-		driver.findElement(MobileBy.AndroidUIAutomator(signUp.checkBoxButtonSelector)).click();
-		
-		
+		signUp.clickTargetUiSelector(signUp.checkBoxButtonSelector);
```
Comment (srahman09) [inline src/test/java/mobiletestrunner/SignUpTest.java:48]: I still think all this should be in the signup form page. 
@codeboostmohshinc thoughts?

**Review by srahman09 — APPROVED** [review]

**Review by faheemc93 — APPROVED** [review]

**Review by srahman09 — APPROVED** [review]

**Pull request merged**